include/RAJA/exec-cuda/scan_cuda.hxx

```diff
@@ -72,73 +71,125 @@ namespace detail
 namespace scan
 {
 
+inline cub::CachingDeviceAllocator& getAllocator() {
+  static cub::CachingDeviceAllocator allocator(true);
+  return allocator;
+}
+
 /*!
         \brief explicit inclusive inplace scan given range, function, and
    initial value
 */
-template <typename InputIter, typename Function, typename T>
-void inclusive_inplace(const ::RAJA::cuda_exec_base&,
+template <size_t BLOCK_SIZE, bool Async,
+          typename InputIter, typename Function, typename T>
+void inclusive_inplace(const ::RAJA::cuda_exec<BLOCK_SIZE, Async>&,
                        InputIter begin,
                        InputIter end,
                        Function binary_op,
                        T init)
 {
-  ::thrust::inclusive_scan(::thrust::device, begin, end, begin, binary_op);
-  cudaDeviceSynchronize();
+  int num_items = std::distance(begin, end);
+  // Determine temporary device storage requirements
+  void     *d_temp_storage = NULL;
+  size_t   temp_storage_bytes = 0;
```

Comment:
These two lines are aligned weird - I don't think it's necessary. Same everywhere these same lines appear.